spring-integration-redis/src/main/java/org/springframework/integration/redis/outbound/RedisStreamMessageHandler.java

```diff
@@ -0,0 +1,90 @@
+package org.springframework.integration.redis.outbound;
+
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.connection.stream.ObjectRecord;
+import org.springframework.data.redis.connection.stream.StreamRecords;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.core.StreamOperations;
+import org.springframework.data.redis.serializer.RedisSerializer;
+import org.springframework.data.redis.serializer.StringRedisSerializer;
+import org.springframework.expression.Expression;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.integration.handler.AbstractMessageHandler;
+import org.springframework.messaging.Message;
+import org.springframework.util.Assert;
+
+/**
+ * Take a message from a Channel and publish it to the given Redis Stream
+ */
+public class RedisStreamMessageHandler extends AbstractMessageHandler {
+
+	private final StreamOperations streamOperations;
+
+	private volatile RedisSerializer<?> serializer = StringRedisSerializer.UTF_8;
+
+	private volatile Expression streamKeyExpression;
+
+	private volatile boolean extractPayload = true;
+
+	public RedisStreamMessageHandler( String streamKey, RedisConnectionFactory connectionFactory ) {
+		Assert.notNull( streamKey, "'streamName' must not be null" );
+		Assert.notNull( connectionFactory, "'connectionFactory' must not be null" );
+		RedisTemplate<String, Object> template = new RedisTemplate<>();
+		template.setConnectionFactory( connectionFactory );
+		template.setValueSerializer( this.serializer );
+		template.setHashValueSerializer( this.serializer );
+		template.setKeySerializer( StringRedisSerializer.UTF_8 );
+		template.setHashKeySerializer( StringRedisSerializer.UTF_8 );
+		template.afterPropertiesSet();
+		this.streamOperations = template.opsForStream();
+		this.setStreamKey( streamKey );
+	}
+
+	public void setSerializer( RedisSerializer<?> serializer ) {
+		Assert.notNull( serializer, "'serializer' must not be null" );
+		this.serializer = serializer;
+	}
+
+	public void setStreamKey( String streamKey ) {
+		Assert.hasText( streamKey, "'streamKey' must not be an empty string." );
+		this.streamKeyExpression = new LiteralExpression( streamKey );
+	}
+
+	public void setStreamKeyExpression( Expression streamKeyExpression ) {
+		this.streamKeyExpression = streamKeyExpression;
```

Comment:
This cannot be `null`. So, an appropriate assert.

Also I think this is really an important option, so has to be in the ctor.
I'm OK with an existing ctor for the `String streamKey`, but then we need to have a new one based on the ` Expression streamKeyExpression`.
Therefore `final` for expression property and not setters at all.